src/components/Button/Button.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import {
+  ActivityIndicator,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+import {ButtonProps} from './ButtonProps.types';
+import {scale, verticalScale} from '../../utils';
+import {useThemeColor} from '../../hooks';
+import Icon from 'react-native-vector-icons/Ionicons';
+
+export const Button = (props: ButtonProps) => {
+  const {
+    children,
+    mode,
+    isLoading,
+    style,
+    textStyle,
+    disabled,
+    onPress,
+    icon,
+    iconRight,
+    iconStyle,
+    loaderColor,
+    ...restProps
+  } = props;
+
+  const {colors} = useThemeColor();
+
+  const getStyle = () => {
+    if (mode === 'default') {
+      return;
+    } else if (mode === 'outlined') {
+      return [
+        styles.outlied,
+        {
+          borderColor: colors.primary,
+        },
+      ];
+    } else if (mode === 'text') {
+      return styles.textType;
+    }
+  };
+
+  const getDisabledBtnStyle = () => {
+    if (disabled && mode === 'text') {
```

Comment:
Since disabled is mandatory for all cases, add a if(!disabled) return as the first condition and remove disabled from rest

Reply:
Also, since we need to return for mode text as well, use an || condition along with disabled

Reply:
fixed